pyuvdata/uvdata.py

```diff
@@ -3451,132 +3910,253 @@ def _smart_slicing(self, data, ind1, ind2, indp, **kwargs):
 
         return out
 
-    def get_data(self, *args, **kwargs):
+    def get_data(self, key1, key2=None, key3=None, squeeze='default',
+                 force_copy=False):
         """
-        Function for quick access to numpy array with data corresponding to
-        a baseline and/or polarization. Returns a read-only view if possible, otherwise a copy.
+        Get the data corresonding to a baseline and/or polarization.
 
-        Args:
-            *args: parameters or tuple of parameters defining the key to identify
-                   desired data. See _key2inds for formatting.
-            **kwargs: Keyword arguments:
-                squeeze: 'default': squeeze pol and spw dimensions if possible
-                         'none': no squeezing of resulting numpy array
-                         'full': squeeze all length 1 dimensions
-                force_copy: Option to explicitly make a copy of the data.
-                             Default is False.
-
-        Returns:
-            Numpy array of data corresponding to key.
+        Parameters
+        ----------
+        key1, key2, key3 : int
```

Comment:
Thank you for cleaning this up!
Small note: should the type technically be `int or tuple of ints`?

Reply:
yes, thanks!